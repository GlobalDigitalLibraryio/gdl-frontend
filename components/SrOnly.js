// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import styled from 'react-emotion';
import { hideVisually } from 'polished';

// A component whose content is only visible to screen readers
const SrOnly = styled('span')(hideVisually);

export default SrOnly;
