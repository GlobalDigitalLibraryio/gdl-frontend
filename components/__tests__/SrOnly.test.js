// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import SrOnly from '../SrOnly';

test('Renders', () => {
  const tree = shallow(<SrOnly>Screen reader only</SrOnly>);

  expect(toJson(tree)).toMatchSnapshot();
});
