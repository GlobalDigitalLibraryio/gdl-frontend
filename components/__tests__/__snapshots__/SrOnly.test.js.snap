// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
.emotion-0 {
  border: 0px;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0px;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<span
  className="emotion-0"
>
  Screen reader only
</span>
`;
